src/settings/rageshake.ts

```diff
@@ -37,37 +38,58 @@ limitations under the License.
 //    actually timestamps. We then purge the remaining logs. We also do this
 //    purge on startup to prevent logs from accumulating.
 
-// the frequency with which we flush to indexeddb
 import { logger } from "matrix-js-sdk/src/logger";
 
+// the frequency with which we flush to indexeddb
 const FLUSH_RATE_MS = 30 * 1000;
 
 // the length of log data we keep in indexeddb (and include in the reports)
 const MAX_LOG_SIZE = 1024 * 1024 * 5; // 5 MB
 
 // A class which monkey-patches the global console and stores log lines.
+
+interface LogEntry {
+  id: string;
+  lines: string;
+  index?: number;
+}
+
+interface Cursor {
+  id: string;
+  ts: number;
+}
+
 export class ConsoleLogger {
   logs = "";
 
-  monkeyPatch(consoleObj) {
+  monkeyPatch(consoleObj: Console): void {
     // Monkey-patch console logging
-    const consoleFunctionsToLevels = {
+
+    const consoleFunctionsToLevels: { [level: string]: string } = {
       log: "I",
       info: "I",
       warn: "W",
       error: "E",
     };
-    Object.keys(consoleFunctionsToLevels).forEach((fnName) => {
-      const level = consoleFunctionsToLevels[fnName];
-      const originalFn = consoleObj[fnName].bind(consoleObj);
-      consoleObj[fnName] = (...args) => {
-        this.log(level, ...args);
-        originalFn(...args);
-      };
-    });
+
+    Object.keys(consoleFunctionsToLevels).forEach(
+      (fnName: "log" | "info" | "warn" | "error") => {
+        const level = consoleFunctionsToLevels[fnName];
+        const originalFn = consoleObj[fnName].bind(consoleObj);
+        consoleObj[fnName] = (...args: unknown[]) => {
+          this.log(level, ...args);
+          originalFn(...args);
+        };
+      }
+    );
   }
 
-  log(level, ...args) {
+  // these functions get overwritten by the monkey patch
+  error(...args: unknown[]): void {}
+  warn(...args: unknown[]): void {}
+  info(...args: unknown[]): void {}
```

Comment:
is there a better way?

Reply:
I'm not immediately seeing why these need to be defined here, but this file is basically taken from react-sdk so I'd just update it to steal the typescript version: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/rageshake/rageshake.ts

Reply:
It is defined so that `this.logger.error(err);` does not raise an error:
`Property 'error' does not exist on type 'ConsoleLogger'.`

Reply:
Okay. Thanks for sharing the react-sdk/rageshake file. It seems it should not use this.logger (the `ConsoleLogger` but instead the imported `PrefixedLogger` (without `.this`).
I will use most of the react-sdk version then.